code/modules/power/gourmonger.dm

```diff
@@ -0,0 +1,276 @@
+		var/nToAdd = min(100, nToDiv)
+		var/obj/item/weapon/reagent_containers/food/snacks/firstMeal = drop_meat(loc)
+		firstMeal.reagents.add_reagent(NUTRIMENT, nToAdd)
+		nToDiv -= nToAdd
+		if(!nToDiv)
+			break
+	meat_amount = 0
+
+/mob/living/simple_animal/hostile/gourmonger/AttackingTarget()
+	if(currentlyMunching)
+		return
+	if(istype(target, /obj/item/weapon/reagent_containers/food/snacks))
+		eatFood(target)
+	if(isliving(target))
+		var/mob/living/M = target
+		if(M.isDeadorDying())
+			eatCorpse(M)
+		else if(hangry)
+			radBurst(kcalPower*2)	//Inherently limited by GOURMONGER_SATISFIED
+			..()
+
+/mob/living/simple_animal/hostile/gourmonger/proc/munchOn(var/T, var/munchTime)
+	flick("gourmonger_eat", src)
+	currentlyMunching = TRUE
+	canmove = 0
+	sleep(munchTime)	//do_after was overkill, especially if there's a lot of them
+	currentlyMunching = FALSE
+	canmove = 1
+	if(Adjacent(T) && !stat)
+		return TRUE
+
+/mob/living/simple_animal/hostile/gourmonger/proc/eatFood(var/obj/item/weapon/reagent_containers/food/snacks/toEat)
+	if(!munchOn(toEat, 3 SECONDS))
+		return
+	var/nValue = 0
+	for(var/datum/reagent/R in toEat.reagents.reagent_list)
+		nValue += (R.nutriment_factor / R.custom_metabolism) * R.volume	//Essentially the nutrition a human would get from 1u
+	toEat.reagents.clear_reagents()
+	eatOutcome(nValue, nValue/2)
+	toEat.after_consume(src)
+
+/mob/living/simple_animal/hostile/gourmonger/proc/eatCorpse(var/mob/living/L)
+	if(ishuman(L))
+		eatLimb(L)
+	else
+		if(!munchOn(L, 3 SECONDS))
+			return
+		var/nValue = decideNutrition(L)
+		eatOutcome(nValue, nValue, 2)
+		L.gib()
+
+/mob/living/simple_animal/hostile/gourmonger/proc/eatLimb(var/mob/living/carbon/human/H)
+	if(!munchOn(H, 3 SECONDS))
+		return
+	var/nValue = decideNutrition(H)
+	var/datum/organ/external/toEat = H.pick_usable_organ(LIMB_LEFT_ARM, LIMB_RIGHT_ARM, LIMB_LEFT_LEG, LIMB_RIGHT_LEG)
+	if(toEat)
+		toEat.droplimb(1, 0, 0)
+		visible_message("<span class='warning'>\The [src] is devouring [H]'s [toEat.display_name]!</span>")
+	else
+		visible_message("<span class='warning'>\The [src] has devoured [H] completely!</span>")
+		H.gib()
+	eatOutcome(nValue, nValue, 3)
+
+/mob/living/simple_animal/hostile/gourmonger/proc/decideNutrition(var/theMeal)
+	var/mealValue = 0
+	if(iscarbon(theMeal))
+		var/mob/living/carbon/C = theMeal
+		mealValue = C.nutrition/5	//arms, legs, and torso since the head isn't eaten
+	if(isanimal(theMeal))
+		var/mob/living/simple_animal/A = theMeal
+		mealValue = A.maxHealth
+	return mealValue
+
+/mob/living/simple_animal/hostile/gourmonger/proc/eatOutcome(var/passToRad, var/passToKcal, var/growMod = 1)
+	radBurst(passToRad)
+	kcalPower += passToKcal
+	growToSplit -= growMod
+	if(growToSplit <= 0)
+		gourSplit()
+
+/mob/living/simple_animal/hostile/gourmonger/proc/sniffForPrey()
+	var/sniffMeal = null
+	var/sniffDist = 50	//Easy way to make 50 the max range and avoids needing to use get_dist twice per loop
+	for(var/client/C in clients)
```

Comment:
I would personally use `player_list` instead but this is about the same order of magnitude.